File: android/src/main/kotlin/co/quis/flutter_contacts/properties/Note.kt
package co.quis.flutter_contacts.properties

data class Note(
    var note: String
) {
    companion object {
        fun fromMap(m: Map<String, Any>): Note = Note(m["note"] as String)
    }

    fun toMap(): Map<String, Any> = mapOf("note" to note)
}

File: ios/Classes/properties/Note.swift
import Contacts

@available(iOS 9.0, *)
struct Note {
    var note: String

    init(fromMap m: [String: Any]) {
        note = m["note"] as! String
    }

    init(fromContact c: CNContact) {
        note = c.note
    }

    func toMap() -> [String: Any] { [
        "note": note,
    ]
    }

    func addTo(_ c: CNMutableContact) {
        c.note = note
    }
}
